In [123]:
from openbb_terminal.sdk import openbb

In [197]:
import pandas as pd
import re

def color_negative_red(valin):
    try:
        val = float(valin.replace(",", ""))
        if val > 0:
            color = 'lightgreen'
        elif val < 0:
            color = 'red'
        else:
            color = 'yellow'
    except:
        try:
            val = float(valin.split(" ")[0].replace(",", ""))
            if val > 0:
                color = 'lightgreen'
            elif val < 0:
                color = 'red'
            else:
                color = 'yellow'
        except:
            color = 'magenta'

    return 'color: %s' % color

def color_dataframe(df: pd.DataFrame):
    """Color the dataframe based on the values of the columns and rows

    Returns
    -------
    df: pd.DataFrame
        colored dataframe
    """
    '''
    for col in df.columns:
        # checks whether column exists
        if col in df.columns:
            df[col] = df[col].apply(lambda x: return_colored_value(str(x)))
   
    for row in df.rows:
        # checks whether row exists
        if row in df.index:
            df.loc[row] = df.loc[row].apply(
                lambda x: return_colored_value(str(x))
            )
    '''
    df.index = [' '.join(re.split('(?<=.)(?=[A-Z])', val)).capitalize() for val in df.index]
    return df.style.format(precision=0).applymap(color_negative_red)

# COINBASE RESEARCH

In [2]:
ticker = "COIN"

In [42]:
openbb.stocks.fa.overview("COIN").T["Description"].values[0]

'Coinbase Global, Inc. provides financial infrastructure and technology for the crypto economy. The company is headquartered in Wilmington, Delaware.'

In [3]:
df = openbb.stocks.load(ticker, start_date="2021-01-01")

Loading Daily data for COIN with starting period 2021-04-14.

In [4]:
openbb.stocks.candle(data=df, symbol=ticker)

In [239]:
for idx, row in openbb.news("Coinbase").reset_index().iterrows():
    print(f'{row["published"]} - {row["title"]}')
    print(row["link"] + "\n")

    if idx > 4:
        break

Fetching data. Please be patient

2023-02-20 09:57:27+00:00 - Does the Arrival of TMS Network (TMSN) Mark the Death Sentence ... - Tekedia
https://news.google.com/rss/articles/CBMieGh0dHBzOi8vd3d3LnRla2VkaWEuY29tL2RvZXMtdGhlLWFycml2YWwtb2YtdG1zLW5ldHdvcmstdG1zbi1tYXJrLXRoZS1kZWF0aC1zZW50ZW5jZS1mb3ItYmluYW5jZS1idXNkLWFuZC1jb2luYmFzZS11c2RjL9IBAA?oc=5

2023-02-20 09:42:00+00:00 - ARK Invest's Coinbase Purchase During Week of Feb 13-17 Totaled 133,321 Shares - CoinDesk
https://news.google.com/rss/articles/CBMieWh0dHBzOi8vd3d3LmNvaW5kZXNrLmNvbS9tYXJrZXRzLzIwMjMvMDIvMjAvYXJrLWludmVzdHMtY29pbmJhc2UtcHVyY2hhc2UtZHVyaW5nLXdlZWstb2YtZmViLTEzLTE3LXRvdGFsZWQtMTMzMzIxLXNoYXJlcy_SAYgBaHR0cHM6Ly93d3cuY29pbmRlc2suY29tL21hcmtldHMvMjAyMy8wMi8yMC9hcmstaW52ZXN0cy1jb2luYmFzZS1wdXJjaGFzZS1kdXJpbmctd2Vlay1vZi1mZWItMTMtMTctdG90YWxlZC0xMzMzMjEtc2hhcmVzLz9vdXRwdXRUeXBlPWFtcA?oc=5

2023-02-20 08:27:49+00:00 - Cryptocurrency Services Market to See Booming Growth | Bitmain ... - Digital Journal
https://news.google.com/rss/articles/CBMicmh0dHBzOi8

I like to start with the company news, to have an idea of what is going on and if there's any red flag or any upcoming event.

In [22]:
openbb.stocks.fa.pt("COIN").tail(5)

Company Rating  Price Target
Date                                                   
2023-02-10  Needham & Company, LLC    BUY          73.0
2023-02-14           Piper Sandler    BUY          65.0
2023-02-14        Barclays Capital   HOLD          57.0
2023-02-16     D.A. Davidson & Co.   HOLD          60.0
2023-02-17           Compass Point    BUY         100.0

In [33]:
print(f'Last adjusted close price {df.tail(1)["Adj Close"].values[0]:.2f}')

Last adjusted close price 65.20


Looking at the last 5 analyst price targets, which date from last week - the stock price is around those values and is either a HOLD or BUY. In fact, the latest price target, from Compass Point, puts stock target at $ 100.

In [34]:
openbb.stocks.fa.pt_chart("COIN")

Loading Daily data for COIN with starting period 2021-04-14.

Looking for price targets over the past year, it's clear that analysts have been rather optimistic about Coinbase. And the stock tends to be below that target price.

In [164]:
openbb.stocks.fa.earnings(ticker)

Earnings Date EPS Estimate Reported EPS Surprise(%)
0    2023-02-21        -2.54            -           -
2    2022-11-03         -2.4        -2.43     -0.0121
3    2022-08-09        -2.65        -4.95     -0.8651
4    2022-05-10         0.21        -1.98    -10.3396
5    2022-02-24         1.85         3.32      0.7936
6    2021-11-09         1.57         1.62      0.0351
7    2021-08-10         2.33         6.42      1.7577
8    2021-05-13         3.09         3.05     -0.0117
9    2021-03-23            -            -           -

In [194]:
openbb.stocks.fa.epsfc(ticker)

fiscalyear consensus_mean    change % analysts  actual consensus_low  \
0       2021         14.665           0     15.0  17.102        10.760   
0       2022         -5.552 -132.463585      8.0       0       -12.620   
0       2023         -2.870  -48.314935     10.0       0        -6.320   

  consensus_high  
0         18.210  
0         -1.940  
0          0.910

Coinbase earnings will occur this week.

It seems that the EPS estimate last quarter was pretty accurate, and given the downturn market in crypto it seems that the estimates are pointing to a worse earnings per share than last quarter.

If this estimate is correct, this means that if the company distributed all of its net income for the period, shareholders would receive for each share of stock they own -2.43.

In [198]:
df = openbb.stocks.fa.balance("COIN", source="AlphaVantage", quarterly=True)
color_dataframe(df)
# note that the colors don't correspond to the rows that we are referring to (e.g. liabilities is green because number is positive even though liability implies a "loss")

Balance sheet shows what a company owns and owes. 

- Coinbase has 111.168 B worth of total assets, 105.542 B worth of total liabilities and 5.626 B in total shareholder equity.
- Most assets and liabilities are current, which means that they can be sold or paid within 1 year.
- The company doesn't distribute earnings.

In [175]:
employees = openbb.stocks.fa.profile(ticker).T["fullTimeEmployees"].values[0]
print(f'Number of employees: {employees}')

Number of employees: 3730


In [182]:
df = openbb.stocks.fa.income("COIN", source="AlphaVantage", quarterly=True)
color_dataframe(df)

Income statement shows how much money a company spent and made.
- Gross profit from Coinbase was 148.826 M. This is the amount of income left over after subtracting the cost of producing goods and selling itrs products from its total sales revenue.
- Total revenue was 590.339 M, which means that the cost of revenue was 441.513 M.
- Cost of revenue is different from cost of goods sold (COGS) because the former also includes costs outside of production, such as distribution and marketing. Since COGS was 101.876 M, it means that distribution and marketing cost Coinbase 339.637 M.
- Operating expenses cost Coinbase 1.147 B last quarter. These are expenses that are not directly related with their product. As other companies did in 2022, Coinbase over-hired last year - as seen on cell above they have around 3730 employees - which may be a value hard to justify.
- After taking into account depreciation and interest and income tax expenses, Coinbase net income is -544.635 M.

In [183]:
df = openbb.stocks.fa.cash("COIN", source="AlphaVantage", quarterly=True)
color_dataframe(df)

Cashflow statements shows the money flowing in and out of the company
- Operating cashflow: -953.501 M
- Investing cashflow: -19.078 M
- Financing cashflow: -42.468 M

In [192]:
df = openbb.stocks.fa.ratios(ticker, quarterly=True)
color_dataframe(df[df.columns[::-1]])

# Note that the only reason why I am able to access quarterly ratios here is because I am subscribed to the premium FMP plan
# After signing up with FMP, this is how I added it here
# openbb.keys.fmp(key="MY FMP KEY", persist=True)

Liquidity ratios:
- Current ratio of 1.055. This corresponds to how a business’s current assets can be used to settle current liabilities. This ratio is okish.
- Quick ratio of 0.056. Coinbase is not in a position to meet its current obligations because it has insufficient assets to be liquidated.
- Cash ratio of 0.049. This measures a business’s ability to use cash and cash equivalent to pay off short-term liabilities, which is poor. But note that this is a more conservative measure because only cash and cash equivalent is used - which is company's most liquid assets.

Leverage ratios:
- Debt-to-equity ratio of 18.759. This ratio tells us that there are $ 18.759 from debt for every dollar invested in the company. This seems very risky, as debt is being used to finance business growth.
- Debt ratio is 0.949, which means that for every dollar in total assets, there's 94.9 cents in debt. Coinbase has more assets than debt, but not by much - they are very leveraged.

Efficiency ratios:
- Asset turnover ratio of 0.005 - measures how much net sales are made from average assets.
- Payable turnover ratio of 1.656 - calculates how quickly Coinbase pays its suppliers and creditors.
- Receivables turnover ratio of 1.962 - measure how quickly they turn customers’ invoices into cash.

Profitability ratios:
- Gross margin of 0.827 - measures how much profit a business makes after the cost of goods and services compared to net sales. This is a great margin, typical from SaaS.
- Operating margin of 0.943 - for every dollar of sale, Coinbase is losing 94.3 cents.
- Return on Assets of -0.005 - determines how much profits they generate from total assets or resources, including current and noncurrent assets. 
- Return on Equity of -0.097 - measures how much profit a business generates from shareholders’ equity. This means that investors are losing with their investment.

Market Value ratios:
- P/E Ratio is -6.628 - measures how valuable a company is relative to its book value earnings per share.
- P/B ratio is 2.567 - measures the market's valuation of a company relative to its book value.
- P/S ratio is 24.461 - indicator of the value that financial markets have placed on each dollar of a company’s sales or revenues.
- Enterprise value multiple is -22.177 - used to calculate the value of a company. This means that the company has more cash than it would need to pay off any debt and buy back all its stocks in one go.

Something about that.

In [205]:
# help(openbb.stocks.fa.est)
color_dataframe(openbb.stocks.fa.est(ticker)[2])

These values are in millions, and this corresponds to the estimates for revenues. The estimates point for a 80% decrease on total revenue since 1 year ago.

In [206]:
# help(openbb.stocks.fa.est)
color_dataframe(openbb.stocks.fa.est(ticker)[1])

These values are in millions, and this corresponds to the estimates for earnings. The estimates point for Coinbase not generating profit anytime soon.

In [209]:
openbb.stocks.fa.mgmt(ticker);
# Business insider doesn't contain information on the management team

No management information in Business Insider for COIN

In [211]:
openbb.stocks.fa.rating(ticker).head(10)

Rating         DCF      ROE      ROA          DE           PE  \
date                                                                       
2023-02-17    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-16    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-15    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-14    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-13    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-10    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-09    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-08    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-07    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   
2023-02-06    Buy  Strong Buy  Neutral  Neutral  Strong Buy  Strong Sell   

                    PB  
date                    
2023-02-17  Strong Buy  
2023-02-16  Strong Buy  
2023-02-15  Strong Buy  
2023-02-14  Strong Buy  
2023-02-13  Strong Buy  
2023-02-10  Strong Buy  
2023-02-09  Strong Buy  
2023-02-08  Strong Buy  
2023-02-07  Strong Buy  
2023-02-06  Strong Buy

With those financials, analyst consensus overall is a Buy. Only when looking to Price to Earnings, the outlook is more worrying where consensus is a strong sell.

In [99]:
openbb.stocks.fa.sec(ticker)

Document Date      Type                 Category Amended  \
Filing Date                                                            
02/16/2023            N/A  SC 13G/A  Institutional Ownership       *   
02/14/2023            N/A  SC 13G/A  Institutional Ownership       *   
02/14/2023            N/A  SC 13G/A  Institutional Ownership       *   
02/10/2023            N/A    SC 13G  Institutional Ownership           
02/09/2023            N/A    SC 13G  Institutional Ownership           
02/01/2023     02/01/2023       8-K           Special Events           
01/10/2023     01/10/2023       8-K           Special Events           
12/02/2022     10/28/2022     8-K/A           Special Events       *   
11/03/2022     09/30/2022      10-Q        Quarterly Reports           
11/03/2022     11/03/2022       8-K           Special Events           
11/02/2022     10/28/2022       8-K           Special Events           
08/09/2022     06/30/2022      10-Q        Quarterly Reports           
08/09/2022     08/09/2022       8-K           Special Events           
06/14/2022     06/14/2022       8-K           Special Events           
06/03/2022     06/01/2022       8-K           Special Events           
05/17/2022     05/16/2022       8-K           Special Events           
05/10/2022     03/31/2022      10-Q        Quarterly Reports           
05/10/2022     05/10/2022       8-K           Special Events           
04/20/2022     06/01/2022   DEF 14A          Proxy Statement           
04/20/2022     04/19/2022       8-K           Special Events           

                                                                                              Link  
Filing Date                                                                                         
02/16/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16412387  
02/14/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16402479  
02/14/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16401805  
02/10/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16385174  
02/09/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16380072  
02/01/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16356117  
01/10/2023   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16308187  
12/02/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16241191  
11/03/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16174965  
11/03/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16174510  
11/02/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16170986  
08/09/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16003699  
08/09/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=16003348  
06/14/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15889596  
06/03/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15873049  
05/17/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15824730  
05/10/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15804000  
05/10/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15803538  
04/20/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15741543  
04/20/2022   https://www.marketwatch.com/investing/stock/coin/financials/secfilings?docid=15741395

When looking at the latest filings, there has been some disclosures around insider trading - which we will look in a second in more detail.

On the 1st of February of 2023, there was a 8K being filed around bylaws amending. However on the 10th of January, one can read:

```
On January 10, 2023, Coinbase Global, Inc. (the “Company”) announced certain preliminary unaudited financial and operating results for its year ended December 31, 2022. (...)

The Company’s audited financial statements for the year ended December 31, 2022 are not yet available. Accordingly, these preliminary financial and operating results are an estimate and subject to the completion of the Company’s financial closing and other procedures and finalization of the Company’s consolidated financial statements for its year ended December 31, 2022, including the completion of the audit of the Company’s financial statements. (...)

The Company further announced that it expects certain of its operating expenses - comprising Sales and Marketing Expenses, Technology and Development + General and Administrative Expenses, including stock-based compensation and excluding restructuring expenses and Other operating expenses, net - for the quarter ending March 31, 2023 to be approximately 25% lower than those operating expenses for the quarter ended December 31, 2022. This is the result of ongoing cost management initiatives, including the Plan (as defined below).

(...)

On January 10, 2023, the Company announced a further restructuring plan (the “Plan”) to manage its operating expenses in response to the ongoing market conditions impacting the cryptoeconomy, as well as ongoing business prioritization efforts. The Plan involves a reduction of the Company’s workforce by approximately 950 employees. The Company expects execution of the Plan to be substantially complete by the second quarter of 2023.

In connection with these actions, the Company estimates that it will incur approximately $149 million to $163 million in total restructuring expenses, consisting of approximately $58 million to $68 million in cash charges related to employee severance and other termination benefits. Of the aggregate charges that the Company expects to incur in connection with the Plan, the Company expects that approximately $91 million to $95 million will be in stock-based compensation expenditures relating to the acceleration of the vesting of outstanding equity awards in accordance with the terms of such awards. The Company expects to recognize substantially all of these charges in the first quarter of 2023.
```

This is a much more interesting statement, which discloses some estimates from Coinbase on their next financials reportings. Worth mentions:
* 950 employees laid off
* Operating expenses reduced by 25%
* 149 to 163 millions in total restructuring expenses (91 to 95 M from it in stock-based compensation, the rest will be in cash charges)

In [213]:
display(openbb.stocks.ins.stats(ticker).head(20)[["Filing Date", "Trading Date", "Insider", "Title", "Trade Type", "Price", "Quantity", "Owned", "Delta Own", "Value", "Filing Link"]])

Filing Date Trading Date          Insider          Title  \
0   2023-02-16 16:11:56   2023-02-14  Armstrong Brian  COB, CEO, 10%   
1   2023-02-09 16:15:54   2023-02-07     Lutke Tobias            Dir   
2   2023-02-06 16:23:08   2023-02-02      Grewal Paul             GC   
3   2023-02-06 16:18:24   2023-02-02  Armstrong Brian  COB, CEO, 10%   
4   2023-02-02 16:20:38   2023-01-31     Lutke Tobias            Dir   
5   2023-02-01 16:13:58   2023-01-30  Armstrong Brian  COB, CEO, 10%   
6   2023-01-26 16:18:30   2023-01-24     Lutke Tobias            Dir   
7   2023-01-20 16:12:21   2023-01-18     Lutke Tobias            Dir   
8   2023-01-18 16:19:47   2023-01-13  Armstrong Brian  COB, CEO, 10%   
9   2023-01-12 16:28:58   2023-01-10    Haas Alesia J            CFO   
10  2023-01-12 16:24:13   2023-01-10     Lutke Tobias            Dir   
11  2023-01-06 16:18:46   2023-01-04  Armstrong Brian  COB, CEO, 10%   
12  2023-01-06 16:15:31   2023-01-04     Lutke Tobias            Dir   
13  2022-12-29 18:53:15   2022-12-28     Lutke Tobias            Dir   
14  2022-12-29 18:50:52   2022-12-27    Haas Alesia J            CFO   
15  2022-12-22 16:10:47   2022-12-20  Armstrong Brian  COB, CEO, 10%   
16  2022-12-22 16:08:22   2022-12-20     Lutke Tobias            Dir   
17  2022-12-15 18:05:37   2022-12-13     Lutke Tobias            Dir   
18  2022-12-15 18:04:00   2022-12-13    Haas Alesia J            CFO   
19  2022-12-08 17:28:57   2022-11-29     Lutke Tobias            Dir   

      Trade Type   Price Quantity    Owned Delta Own        Value  \
0    S - Sale+OE  $57.83  -29,732        0     -100%  -$1,719,338   
1   P - Purchase  $72.95   +4,960  209,352       +2%    +$361,832   
2       S - Sale  $77.06   -1,818   55,424       -3%    -$140,099   
3    S - Sale+OE  $79.33  -44,598        0     -100%  -$3,537,757   
4   P - Purchase  $57.70   +6,588  204,392       +3%    +$380,128   
5    S - Sale+OE  $58.61  -29,732        0     -100%  -$1,742,676   
6   P - Purchase  $54.38   +6,612  197,804       +3%    +$359,561   
7   P - Purchase  $54.21   +6,834  191,192       +4%    +$370,471   
8    S - Sale+OE  $51.02  -89,196        0     -100%  -$4,551,203   
9    S - Sale+OE  $38.53  -88,000  276,464      -24%  -$3,390,469   
10  P - Purchase  $37.76   +9,668  184,358       +6%    +$365,064   
11   S - Sale+OE  $36.84  -14,866        0     -100%    -$547,708   
12  P - Purchase  $34.62  +11,012  174,690       +7%    +$381,235   
13  P - Purchase  $32.58  +11,332  163,678       +7%    +$369,197   
14   S - Sale+OE  $33.83  -94,000  214,464      -30%  -$3,180,054   
15   S - Sale+OE  $35.31  -14,866        0     -100%    -$524,980   
16  P - Purchase  $34.75  +10,520  152,346       +7%    +$365,570   
17  P - Purchase  $46.04   +8,685  141,826       +7%    +$399,857   
18   S - Sale+OE  $45.31  -92,000  158,464      -37%  -$4,168,572   
19  P - Purchase  $44.34  +16,747  133,141      +14%    +$742,514   

                                                                                                  Filing Link  
0   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000023/xslF345X03/wf-form4_167658190072846.xml  
1   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000020/xslF345X03/wf-form4_167597733503531.xml  
2   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000014/xslF345X03/wf-form4_167571857037876.xml  
3   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000013/xslF345X03/wf-form4_167571827577220.xml  
4   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000012/xslF345X03/wf-form4_167537282094650.xml  
5   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000010/xslF345X03/wf-form4_167528601969484.xml  
6   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000008/xslF345X03/wf-form4_167476789573526.xml  
7   http://www.sec.gov/Archives/edgar/data/1679788/000167978823000007/xslF345X03/wf-form4_167424912758577.xml  
8   http://www.sec.gov/Archives/edgar/dat

The SEC filings from Coinbase employees trading stock is interesting. While Brian (CEO) has been selling, Tobias Lutke (Director) has been accumulating for the past few months.

In [224]:
insider = "Lutke Tobias"

df_insider = openbb.stocks.ins.stats(ticker)[["Filing Date", "Trading Date", "Insider", "Title", "Trade Type", "Price", "Quantity", "Owned", "Delta Own", "Value", "Filing Link"]]
total_bought = sum([float(val.replace("$", "").replace(",", "")) for val in df_insider[df_insider["Insider"] == insider]["Value"].values])

print(f'Total bought by {insider}: $ {total_bought/1_000_000:.2f} M')

Total bought by Lutke Tobias: $ 10.07 M


This is a very high value, after some research it's easy to find out that Tobias Lutke is Shopify CEO, and the news reported this event:
https://blockworks.co/news/shopify-ceo-spends-3m-on-coinbase-stock-in-2-months

In [225]:
openbb.stocks.gov.contracts(ticker)

Date                           Description  \
0  2022-05-20  COINBASE ANALYTICS PLATFORM LICENSES   

                                  Agency   Amount  
0  DEPARTMENT OF HOMELAND SECURITY (DHS)  29000.0

Government awarded a contract to Coinbase in May of 2022

In [105]:
openbb.stocks.gov.lobbying(ticker)

Date Ticker          Client     Amount  \
0  2023-01-20   COIN  COINBASE, INC.    80000.0   
1  2023-01-20   COIN  COINBASE, INC.   590000.0   
2  2023-01-20   COIN   COINBASE, INC        0.0   
3  2023-01-20   COIN  COINBASE, INC.    60000.0   
4  2023-01-19   COIN  COINBASE, INC.    30000.0   
5  2023-01-18   COIN  COINBASE, INC.    60000.0   
6  2023-01-13   COIN  COINBASE, INC.    50000.0   
7  2022-10-20   COIN  COINBASE, INC.  1040000.0   
8  2022-10-20   COIN  COINBASE, INC.    70000.0   
9  2022-10-20   COIN  COINBASE, INC.    80000.0   

                                                                                        Issue  \
0                          Computer Industry \n Financial Institutions/Investments/Securities   
1                                               Financial Institutions/Investments/Securities   
2                                                                                        None   
3  Taxation/Internal Revenue Code \n Banking \n Financial Institutions/Investments/Securities   
4                                               Financial Institutions/Investments/Securities   
5                                               Financial Institutions/Investments/Securities   
6                                               Financial Institutions/Investments/Securities   
7                                               Financial Institutions/Investments/Securities   
8  Taxation/Internal Revenue Code \n Banking \n Financial Institutions/Investments/Securities   
9                          Computer Industry \n Financial Institutions/Investments/Securities   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Specific_Issue  
0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         P

In [237]:
openbb.stocks.gov.government_trading(gov_type="congress", symbol=ticker).head(10)

ReportDate TransactionDate Ticker  Representative Transaction   Amount  \
0  2023-02-09      2022-12-20   COIN   Shri Thanedar        Sale   1001.0   
1  2022-12-23      2022-11-21   COIN    Marie Newman        Sale   1001.0   
2  2022-10-18      2022-09-30   COIN   Garret Graves    Purchase   1001.0   
3  2022-02-08      2022-01-04   COIN    Marie Newman    Purchase   1001.0   
4  2022-01-25      2021-12-21   COIN   Ed Perlmutter        Sale   1001.0   
5  2021-12-08      2021-11-18   COIN    Marie Newman    Purchase  15001.0   
6  2021-12-08      2021-11-09   COIN    Marie Newman    Purchase  15001.0   
7  2021-12-09      2021-11-05   COIN    Marie Newman        Sale  50001.0   
8  2021-12-08      2021-11-05   COIN    Marie Newman        Sale  50001.0   
9  2021-10-26      2021-09-23   COIN   Ed Perlmutter    Purchase   1001.0   

             House             Range  
0  Representatives    $1,001-$15,000  
1  Representatives    $1,001-$15,000  
2  Representatives    $1,001-$15,000  
3  Representatives    $1,001-$15,000  
4  Representatives    $1,001-$15,000  
5  Representatives   $15,001-$50,000  
6  Representatives   $15,001-$50,000  
7  Representatives  $50,001-$100,000  
8  Representatives  $50,001-$100,000  
9  Representatives    $1,001-$15,000

Note that these transactions occurred last year, as the representative only needs to financially disclose these within a 30 day window (which can be extended to 45)